### PR TITLE
Convert some methods from DSInstance to update plugins

### PR DESCRIPTION
**Pull request by stlaz**
The DSInstance methods replica_populate and update_dna_shared_config can be easily converted to being update plugins. This should prove useful to generalize installation steps.

### CODE REVIEW

File: ipaserver/install/dsinstance.py [hunk 1]
```diff
@@ -1116,26 +1121,6 @@ def add_extdom_plugin(self, suffix):
         else:
             root_logger.debug("extdom plugin is already configured")
 
-    def replica_populate(self):
```
Comment (jcholast) [inline ipaserver/install/dsinstance.py]: Looks like this belongs into the next commit ("Move ds.replica_populate to an update plugin").

**Comment by jcholast** [comment]
LGTM

**Comment by martbab** [comment]
The code works both during upgrade and during replica install. There are some problems with duplicate entries being created by DNA plugin itself but that is not the problem of FreeIPA IMHO. I guess we can merge this PR and open a FreeIPA tracker for https://fedorahosted.org/389/ticket/48373

**Comment by stlaz** [comment]
+1

**Comment by jcholast** [comment]
@martbab, works for me.

**Comment by martbab** [comment]
ACK

**Comment by jcholast** [comment]
Pushed to installer-refactoring: 5f1befcbe6222084d7a72134d2d9eb6d8305aabe.

**Pull request closed without merging**